presto-hive/src/main/java/io/prestosql/plugin/hive/procedure/SyncPartitionMetadataProcedure.java

```diff
@@ -71,6 +72,7 @@
             SyncPartitionMetadataProcedure.class,
             "syncPartitionMetadata",
             ConnectorSession.class,
+            ConnectorAccessControl.class,
```

Comment:
we should also inject `ConnectorSecurityContext` so it could be used instead of `null`. It could be artificial but user should the one passed by us when checking the access.

Reply:
We discussed this offline and null seems cleaner than passing a fake security context that won't be used. `ConnectorSecurityContext` is a concrete class and doesn't have all the information needed by the access control. Passing null is the same as passing back the same one, but is more clear it won't be used, and is less code on both sides.

Reply:
I see. I completely understand the issue from Presto developer perspective.

However, from Presto user (author of custom Presto plugin) it sounds weird. That in this particular case they should pass `null` for not `@Nullable` or `Optional` parameter. I bet most of them would ask us how to get `ConnectorSecurityContext` to call `ConnectorAccessControl` in such context. On the other hand bets that someone would have this case are low.

Since we already have custom implementation of `ConnectorAccessControl`, why not to have a new entity that would be similar to `ConnectorAccessControl` but without `ConnectorSecurityContext` parameter. Kind of a SPI view of `AccessControl`. Notice that it does not have to be fully featured view on `AccessControl`. We could start with few methods.

Reply:
This is my point of view. I am not going to block anything here because of that.

Disregarding which side we take this is still great improvement.

Reply:
We had also discussed having a custom access control interface (that was my original proposal), but @dain though that was overkill when `ConnectorAccessControl` is usable. Maybe he can explain more.